### PR TITLE
Create AdminUserIdentity

### PR DESCRIPTION
**Pull request by emma-p**
Wraps UserIdentity into a new AdminIdentity to add a constraint on the login (a logged in user should be an admin user and not only a valid `guardian.co.uk` user)

### CODE REVIEW

**Comment by danmassie** [comment]
Why do we need to wrap the UserIdentity in an AdminUserIdentity case class?

**Comment by emma-p** [comment]
We don't need to but I thought in this case it could be useful to use an additional type to enforce the fact that we need an Admin User and not only an Identity User in order to log someone in

**Comment by danmassie** [comment]
All auth is done via google auth for this app and if auth fails, there won't be an instance of UserIdentity. IMHO I don't think its worth it unless we are extending com.gu.googleauth.UserIdentity in some way.

File: app/controllers/Login.scala [hunk 1]
```diff
@@ -46,16 +47,16 @@ class Login @Inject() (conf: Config) extends Controller with AuthActions with Lo
 
     type FutureEither[A] = EitherT[Future, LoginError, A]
 
-    val result: FutureEither[UserIdentity] = for {
-      identity <- EitherT[Future, LoginError, UserIdentity](validateUser(authConfig))
-      validUserAdmin <- EitherT[Future, LoginError, UserIdentity](GoogleGroups.isUserAdmin(identity))
+    val result: FutureEither[AdminUserIdentity] = for {
```
Comment (markjamesbutler) [inline app/controllers/Login.scala]: This makes the code much more readable.

File: app/controllers/Login.scala [hunk 2]
```diff
@@ -46,16 +47,16 @@ class Login @Inject() (conf: Config) extends Controller with AuthActions with Lo
...
+      identity <- EitherT(validateUser(authConfig))
+      validUserAdmin <- EitherT(GoogleGroups.isUserAdmin(identity))
```
Comment (markjamesbutler) [inline app/controllers/Login.scala]: Let's rename validUserAdmin to userAdmin as AdminUserIdentity will always be valid.

File: app/model/AdminUserIdentity.scala [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package model
+
+import com.gu.googleauth.UserIdentity
+
+case class AdminUserIdentity(userIdentity: UserIdentity)
```
Comment (markjamesbutler) [inline app/model/AdminUserIdentity.scala]: AdminUser or AdminIdentity might be a better name.

**Comment by markjamesbutler** [comment]
:+1: Please add description to explain intent of this PR.

**Pull request merged**